commit/merkleroot/rmn/controller.go

```diff
@@ -637,20 +648,31 @@ func getMostVotedRootsFromObservations(
 		}
 	}
 
-	mostVotedRoots := make(map[cciptypes.ChainSelector]cciptypes.Bytes32)
+	selectedRoots := make(map[cciptypes.ChainSelector]cciptypes.Bytes32)
 	for chain, votes := range votesPerRoot {
-		var mostVotedRoot cciptypes.Bytes32
-		maxVotes := 0
+		minObserversForChain, exists := minObservers[chain]
+		if !exists {
+			return nil, fmt.Errorf("no min observers for chain %d", chain)
+		}
+
+		var selectedRoot cciptypes.Bytes32
+
 		for root, vote := range votes {
-			if vote > maxVotes {
-				mostVotedRoot = root
-				maxVotes = vote
+			if vote < minObserversForChain {
+				continue
 			}
+
+			if !selectedRoot.IsEmpty() {
+				return nil, fmt.Errorf("more than one valid root for chain %d", chain)
+			}
+
+			selectedRoot = root
 		}
-		mostVotedRoots[chain] = mostVotedRoot
+
+		selectedRoots[chain] = selectedRoot
```

Comment:
Is it adding an empty root is no root is valid?

If yes, we might want to add an non-empty check before adding it to the selectedRoots

```
if selectedRoot.IsEmpty() {
    return nil, fmt.Errorf("no valid root found for chain %d", chain)
}
```